src/service_discovery.rs

```diff
@@ -176,9 +176,12 @@ impl<Exe: Executor> ServiceDiscovery<Exe> {
         &self,
         topic: S,
     ) -> Result<u32, ServiceDiscoveryError> {
-        let mut connection = self.manager.get_base_connection().await?;
         let topic = topic.into();
+        if topic.contains("-partition-") {
```

Comment:
maybe a comment to explain where does come from this magic string

Reply:
yes, of course